server/controllers/credit_card_controller.py

```diff
@@ -14,6 +14,15 @@ def get_credit_card_by_number(self, card_number):
         credit_card = CreditCard.query.filter_by(card_number=card_number).first()
         return jsonify(credit_card.to_dict() if credit_card else {})
 
+    def get_credit_card_balance(self, card_number):
+        credit_card = CreditCard.query.filter_by(card_number=card_number).first()
```

Comment:
minor thing: can you jsut do get(card_number=card_number) and that returns if found? if not, curent method fine.